### PR TITLE
Adds mocks generator plugin. Closes #305

### LINKED ISSUE #305: Create mocks from intercepted requests
Create a new summary plugin that creates mocks from the intercepted requests. This will allow users to create mocks based on their data.

After generating the mocks, the plugin sorts them descending by URL length to put the most specific URLs upfront and more generic URLs in the back. This will allow users to replace IDs with wildcards to generalize their mocks and ensure that they will execute in the correct order.

### PR DESCRIPTION
**Pull request by waldekmastykarz**
Adds mocks generator plugin. Closes #305

### CODE REVIEW

**Comment by waldekmastykarz** [comment]
Let's merge after releasing v0.11

File: m365-developer-proxy-plugins/MockResponses/MockResponsePlugin.cs [hunk 1]
```diff
@@ -16,7 +16,9 @@
 namespace Microsoft365.DeveloperProxy.Plugins.MocksResponses;
 
 internal class MockResponseConfiguration {
+    [JsonIgnore]
     public bool NoMocks { get; set; } = false;
+    [JsonIgnore]
```
Comment (gavinbarron) [inline m365-developer-proxy-plugins/MockResponses/MockResponsePlugin.cs:21]: won't this prevent passing the location of the mock file via config?

Comment (waldekmastykarz) [reply m365-developer-proxy-plugins/MockResponses/MockResponsePlugin.cs:21]: The way I understood it, is that it's for serialization, but let me double check that it doesn't affect deserialization. Good catch

Comment (waldekmastykarz) [reply m365-developer-proxy-plugins/MockResponses/MockResponsePlugin.cs:21]: Just verified and it doesn't prevent configuring the mocks file via config

**Review by gavinbarron — APPROVED** [review]
LGTM

**Pull request merged**